src/ssl.c

```diff
@@ -14072,6 +14072,22 @@ static WC_INLINE void RestoreSession(WOLFSSL* ssl, WOLFSSL_SESSION* session,
 #endif
 }
 
+static int SslSessionCacheOn(const WOLFSSL* ssl, const WOLFSSL_SESSION* session)
+{
+    (void)session;
+    if (ssl->options.sessionCacheOff
+    #if defined(HAVE_SESSION_TICKET) && defined(WOLFSSL_FORCE_CACHE_ON_TICKET)
+                && session->ticketLen == 0
+    #endif
+    #ifdef OPENSSL_EXTRA
+                && ssl->options.side != WOLFSSL_CLIENT_END
+    #endif
+                )
+        return WOLFSSL_FAILURE;
+    else
+        return WOLFSSL_SUCCESS;
```

Comment:
Please re-write with single exit. Please use 0 success and -1 for failure, not compatibility return codes.

Reply:
Rewritten with single exit and using boolean return.